dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/outline/OutlineTypeParser.java

```diff
@@ -107,10 +109,24 @@ public void visit(
       typeOutline = new TypeOutline(version, access, name, superName, interfaces);
     }
 
+    @Override
+    public void visitOuterClass(String owner, String name, String descriptor) {
+      if (name != null && !name.equals(MethodDescription.TYPE_INITIALIZER_INTERNAL_NAME)) {
+        selfContained = false;
+      } else if (owner != null) {
+        selfContained = false;
+      }
+    }
+
     @Override
     public void visitInnerClass(String name, String outerName, String innerName, int access) {
-      if (null != outerName && typeOutline.getInternalName().equals(name)) {
-        typeOutline.declaredBy(outerName);
+      if (typeOutline.getInternalName().equals(name)) {
+        if (null != outerName) {
+          typeOutline.declaredBy(outerName);
+        }
+        if (null == innerName && !selfContained) {
```

Comment:
```suggestion
        else if (null == innerName && !selfContained) {
```
because if we know the type is declared-by another type then it can't be anonymous?

Reply:
Yep you are right, fixed, thanks!